### DISCUSSION TITLE
Is HttpClientFactory the safer option?

### DISCUSSION THREAD

**Discussion by NickLydon** [canton7/RestEase#203] [Q&A] [answered]
There were previously two issues with creating and disposing HttpClient directly - recognising DNS changes and socket exhaustion. 
[This stackoverflow Q+A](https://stackoverflow.com/a/58428122) discusses them. I want to know whether these concerns with HttpClient are still valid with .NET 5.
1. Is there a reason to use HttpClientFactory other than for extra functionality, e.g. with Polly integration?
2. Is it safe to create a new HttpClient/RestEase client for each usage?
3. Is it safe to use a singleton HttpClient/RestEase client? 

Thanks for your help, and apologies if this has already been answered.

**Comment by canton7** [comment, marked as answer]
My understanding is that [`SocketsHttpClientHandler`](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.socketshttphandler?view=net-5.0#remarks), introduced in .NET Core 2.1, basically fixes all of those issues, so it doesn't matter whether you create lots of new HttpClients, or cache a single one.

I don't *think* there's any strong need to use HttpClientFactory therefore, although it does tie nicely in with the rest of that infrastructure (e.g. chaining in order HttpMessageHandlers, Polly, etc etc) which means it makes sense to use it if it's available.
